apps/meteor/app/utils/lib/mimeTypes.ts

```diff
@@ -12,7 +14,14 @@ const getExtension = (param: string): string => {
 	return !extension || typeof extension === 'boolean' ? '' : extension;
 };
 
-const getMimeType = (fileName: string): string => {
+const getMimeType = (mimetype: string, fileName: string): string => {
+	// If the extension from the mimetype is different from the file extension, the file extension may be wrong
+	// so use the informed mimetype
```

Comment:
I believe we'd have another issue with this approach because `mimetype` is not always correct. We may receive `application/octet-stream` from it, which could mess things up

Let's use `lst` as an example (which I tried locally): [this line](https://github.com/RocketChat/Rocket.Chat/pull/32763/files#diff-b3f6491decaee72b3d324f149c851d11cb1288ec73b30cf1641baa9edefb397aR88) will provide `application/octet-stream` to this function when `fileName` is `test.lst`. Since `mime.extension('application/octet-stream')` returns `bin`, we'd use `application/octet-stream` as the MIME type for this file, which is incorrect

Reply:
We need to trust on something. The mime type is much more reliable than extension.

Reply:
I think it looks better with the changes you applied now, it should be ok to rely on the mime type (since this onde may be extracted by analyzing more info than just the file name) **when** it's not `application/octet-stream`